stripped/colabs/Regresja_logistyczna_i_SVM_stripped.py

```diff
@@ -0,0 +1,1180 @@
+          fold.append(blocks_list[r])
+          blocks_list.remove(blocks_list[r])
+      folds.append(fold)
+
+  if reminder!=0:
+     for b in blocks_list:
+        if random_state:
+           random.seed(random_state)
+        n_fol = random.randint(0, k-1)
+        folds[n_fol].append(b)
+        blocks_list.remove(b)
+
+  foldsDF_list = []
+  folds_data = [df[df['block'].isin(fold)] for fold in folds]
+  foldsDF_list = [pd.DataFrame(data=fold_data) for fold_data in folds_data]
+  foldsDF_list = [foldDF.drop(columns=['block']) for foldDF in foldsDF_list]
+
+  return foldsDF_list
+
+ min_lat = 25.0625
+ max_lat = 52.9375
+ min_lon = -124.9375
+ max_lon = -67.0625
+
+blocks = block_partition(July_2019_DF, 2.5, min_lat, max_lat, min_lon, max_lon)
+folds = Kfolds(blocks, 6, random_state=3)
+
+for f in folds:
+  print(len(f))
+  print(len(f[f['pustynia']==1]))          #dla random_state=3 otrzymujemy podział z rozsadna iloscia pustyn w kazdym foldzie
+
+def test_for_residuals_normality(df: pd.DataFrame, column_name:str):
+
+  r1 = df[df['pustynia']==1][column_name] - df[df['pustynia']==1][column_name].mean()
+  r2 = df[df['pustynia']==0][column_name] - df[df['pustynia']==0][column_name].mean()
+  x = list(r1) + list(r2)
+  shapiro = stats.shapiro(x)
+  print(shapiro)
+  sm.qqplot(np.array(x), line='s')
+
+def OLS_report_and_ANOVA(df: pd.DataFrame, group_column_name: str, feature_column_name: str) -> None:
+
+  groups = pd.DataFrame({ group_column_name: df[group_column_name],
+                         feature_column_name: df[feature_column_name]})
+  model = sm.OLS(groups[feature_column_name], groups[group_column_name]).fit()
+  print(model.summary())
+
+  groups2 = pd.DataFrame({ 'group': df[group_column_name], 'feature': df[feature_column_name]})
+  model2 = ols(formula="feature~group", data=groups2).fit()   #anova_lm requires formula.api.ols instead of api.OLS
+  anova = sm.stats.anova_lm(model2, typ=2)
+  print(' ')
+  print('ANOVA results:')
+  print (anova)
+
+
+def plot_density(df: pd.DataFrame, group_column_name: str, feature_column_name: str, random_state: Optional[int]=13) -> None:
+
+  df_groups = pd.DataFrame({'Desert': df[df[group_column_name]==1][feature_column_name],
+                 'Non-desert': df[df[group_column_name]==0][feature_column_name]})
+  random.seed(random_state)
+  desert_sample = random.sample(list(df_groups['Desert'].dropna()), 50)
+  random.seed(random_state)
+  nondesert_sample = random.sample(list(df_groups['Non-desert'].dropna()), 50)
+
+  samples = pd.DataFrame({'Desert': desert_sample, 'Non-desert': nondesert_sample})
+  samples.plot.density()
+
+
+def perform_Kruskal_Wallis(df: pd.DataFrame, group_column_name:str, feature_column_name: str, random_state: Optional[int] =13) -> None:
+
+  df_groups = pd.DataFrame({'Desert': df[summer_data[group_column_name]==1][feature_column_name],
+                 'Non-desert': df[summer_data[group_column_name]==0][feature_column_name]})
```

Comment:
summer data to jakaś przecieknięta zmienna czy literówka? 
Poza tym ta 0 i 1 wygląda jakoś słabo pod względem konfiguracji. Sugeruje parametr z wartościami domyślnymi w tym przypadku np. label_for_desert

Reply:
Ojć, tak, literówka sie wkradła! Baardzo dziękuję za czujność! Powinno być "df" oczywiście.